config/prow/config.yaml

```diff
@@ -523,6 +523,10 @@ branch-protection:
           required_status_checks:
             contexts:
             - lint
+        cluster-api-provider-vsphere:
+          required_status_checks:
+            contexts:
```

Comment:
This makes golangci-lint mandatory for merge. Otherwise it's easy to miss in case the action is not run automatically for new contributors (someone with write access has to approve the action in this case)